### PR TITLE
feat: add support of custom page/screen event name in mixpanel

### PR DESCRIPTION
**Pull request by Gauravudia**
## What are the changes introduced in this PR?

- Added the support for page/screen custom event name
- `generatePageOrScreenCustomEventName` utility -> Generates a custom event name for a page or screen with given event handlebar
- Introduce 2 new toggle for page/screen events 
  - useUserDefinedPageEventName - Toggle to decide whether to send a custom page event name
  - useUserDefinedScreenEventName - Toggle to decide whether to send custom screen event name
- Introduce 2 settings for page/screen event template
  -  userDefinedPageEventTemplate
  -  userDefinedScreenEventTemplate

## What is the related Linear task?

Resolves INT-1553

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [x] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [x] Is the type of change in the PR title appropriate as per the changes?

- [x] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3098/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3098/test-report.html

File: src/v0/destinations/mp/util.test.js [hunk 1]
```diff
@@ -98,371 +147,338 @@ describe('Mixpanel utils test', () => {
...
+    expect(result).toBe(expected);
+  });
+
+  it('should return a event name when message object is not provided', () => {
+    const message = undefined;
+    const userDefinedEventTemplate = 'Viewed a {{ path }} page';
+    const expected = 'Viewed a  page';
+    const result = generatePageOrScreenCustomEventName(message, userDefinedEventTemplate);
+    expect(result).toBe(expected);
+  });
```
Comment (ItsSudip) [inline src/v0/destinations/mp/util.test.js:483]: What will happen if multiple handlebers are there in template? Like if the template is `{{ name }} page viewed from {{ context.page.url.host }}`

Comment (Gauravudia) [reply src/v0/destinations/mp/util.test.js:483]: Added the testcase for it.

File: src/v0/destinations/mp/util.js [hunk 2]
```diff
@@ -301,6 +301,41 @@ function trimTraits(traits, contextTraits, setOnceProperties) {
...
+ * @param {Object} message - The message object
+ * @param {string} userDefinedEventTemplate - The user-defined event template to be used for generating the event name.
+ * @throws {ConfigurationError} If the event template is missing.
+ * @returns {string} The generated custom event name.
+ * @example
+ * const userDefinedEventTemplate = "Viewed {{ category }} {{ name }} Page";
+ * const message = {name: 'Home', properties: {category: 'Index'}};
+ * output: "Viewed Index Home Page"
+ */
+const generatePageOrScreenCustomEventName = (message, userDefinedEventTemplate) => {
```
Comment (saikumarrs) [inline src/v0/destinations/mp/util.js:316]: Please match this logic with the device mode implementation.

Comment (Gauravudia) [reply src/v0/destinations/mp/util.js:317]: updated the code to match with device mode

File: src/v0/destinations/mp/util.js [hunk 3]
```diff
@@ -301,6 +302,46 @@ function trimTraits(traits, contextTraits, setOnceProperties) {
...
+
+  if (isDefinedAndNotNull(message.name)) {
+    // Replace {{ name }} with actual values
+    eventName = eventName.replace(/{{\s*name\s*}}/g, message.name);
+  } else {
+    // find {{ name }} surrounded by whitespace characters and replace it with a single whitespace character
+    eventName = eventName.replace(/\s{{\s*name\s*}}\s/g, ' ');
+  }
+
+  return eventName.trim();
```
Comment (saikumarrs) [inline src/v0/destinations/mp/util.js:342]: We should not trim any leading or trailing whitespaces.

Comment (Gauravudia) [reply src/v0/destinations/mp/util.js]: This was not the case in device mode, Fixed it cloud mode.

**Review by saikumarrs — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**